### PR TITLE
[DF] Fix RDF compilation with gcc < 7.3

### PR DESCRIPTION
**Pull request by eguiraud**
Before gcc 7.3 there is an issue passing a non-movable pair<string, unique_ptr<T>> into unordered_map::insert. A standalone reproducer is at https://godbolt.org/z/1ror6sqPx .

This patch works around it by using operator[] + assignment instead.

### CODE REVIEW

File: tree/dataframe/src/RDFColumnRegister.cxx [hunk 1]
```diff
@@ -40,8 +40,16 @@ RDefineReader *RDefinesWithReaders::GetReader(unsigned int slot, const std::stri
    if (variationName != "nominal")
       define = &define->GetVariedDefine(variationName);
 
+#if !defined(__clang__) && __GNUC__ >= 7 && __GNUC_MINOR__ >= 3
    const auto insertion = defineReaders.insert({variationName, std::make_unique<RDefineReader>(slot, *define)});
-   return insertion.first->second.get();
+   auto *ret = insertion.first->second.get();
```
Comment (Axel-Naumann) [inline tree/dataframe/src/RDFColumnRegister.cxx:45]: Why did you change this? Why not just `return ret` for the old case? Once we remove the old GCC case we would be left with this extra return...

File: tree/dataframe/src/RDFColumnRegister.cxx [hunk 2]
```diff
@@ -63,9 +71,17 @@ RVariationsWithReaders::GetReader(unsigned int slot, const std::string &colName,
    if (it != varReaders.end())
       return it->second.get();
 
+#if !defined(__clang__) && __GNUC__ >= 7 && __GNUC_MINOR__ >= 3
    const auto insertion =
       varReaders.insert({variationName, std::make_unique<RVariationReader>(slot, colName, variationName, *fVariation)});
-   return insertion.first->second.get();
+   auto *ret = insertion.first->second.get();
```
Comment (Axel-Naumann) [inline tree/dataframe/src/RDFColumnRegister.cxx:77]: Same comment here.

**Review by Axel-Naumann — APPROVED** [review]
But see comment

File: tree/dataframe/src/RDFColumnRegister.cxx (hunk 1 above)
Comment (eguiraud) [reply tree/dataframe/src/RDFColumnRegister.cxx:45]: it seems that I have an ingrained dislike for multiple return statements. even when only one gets to be compiled. i'll fix it, one sec.

Comment (eguiraud) [reply tree/dataframe/src/RDFColumnRegister.cxx]: it should be ok now

**Review by vepadulano — APPROVED** [review]
Thanks!
